src/plugins/vis_augmenter/public/utils/utils.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { get } from 'lodash';
+import { Vis } from '../../../visualizations/public';
+import {
+  formatExpression,
+  buildExpressionFunction,
+  buildExpression,
+  ExpressionAstFunctionBuilder,
+} from '../../../../plugins/expressions/public';
+import { ISavedAugmentVis, SavedAugmentVisLoader, VisLayerFunctionDefinition } from '../';
+
+// TODO: provide a deeper eligibility check.
+// Tracked in https://github.com/opensearch-project/OpenSearch-Dashboards/issues/3268
+export const isEligibleForVisLayers = (vis: Vis): boolean => {
+  return vis.params.type === 'line';
+};
+
+export const getAugmentVisSavedObjs = async (
+  visId: string | undefined,
+  loader: SavedAugmentVisLoader | undefined
+): Promise<ISavedAugmentVis[]> => {
+  try {
+    const resp = await loader?.findAll();
+    const allSavedObjects = (get(resp, 'hits', []) as any[]) as ISavedAugmentVis[];
+    return allSavedObjects.filter((hit: ISavedAugmentVis) => hit.visId === visId);
+  } catch (e) {
+    return [] as ISavedAugmentVis[];
+  }
+};
+
+export const buildPipelineFromAugmentVisSavedObjs = (objs: ISavedAugmentVis[]): string => {
+  const visLayerExpressionFns = [] as Array<
+    ExpressionAstFunctionBuilder<VisLayerFunctionDefinition>
+  >;
+
+  try {
+    objs.forEach((obj: ISavedAugmentVis) => {
+      visLayerExpressionFns.push(
+        buildExpressionFunction<VisLayerFunctionDefinition>(
+          obj.visLayerExpressionFn.name,
```

Comment:
*Doubt:* We call this as expressionFn but I am not sure if it has an actual function implementation associated with it because from [the type](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/feature/feature-anywhere/src/plugins/vis_augmenter/public/types.ts#L19) I see it has a name and args associated with it. Can you point me to a place that can help me understand this better? Thanks!

Reply:
Sure - it's defined in `src/plugins/vis_augmenter/public/types.ts` in #3109. Essentially it's an interface that defines how to run a plugin-defined expression function. `type` represents the return type of the expression function (same as existing expression functions), `name` is the name of the registered function, and `args` is the loosely-typed field for plugins to add their own args. An example would be something like an `add_anomalies` function defined in the AD plugin, which would collect anomalies and return the return type.

These functions are then read here and parsed out to create a single expressions string pipeline, where when executed, will collect all `VisLayer`s.

See details and diagrams in the meta issue #2880